lib/shared/image_viewer.dart

```diff
@@ -1,61 +1,75 @@
+import 'dart:io';
+import 'package:image_gallery_saver/image_gallery_saver.dart';
+import 'package:path/path.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:cached_network_image/cached_network_image.dart';
 import 'package:flutter/material.dart';
+import 'package:photo_view/photo_view.dart';
+import 'package:flutter_cache_manager/flutter_cache_manager.dart';
+import 'package:permission_handler/permission_handler.dart';
 
-import 'package:extended_image/extended_image.dart';
-
-import 'package:thunder/shared/hero.dart';
-
-class ImageViewer extends StatefulWidget {
+class ImageViewer extends StatelessWidget {
   final String url;
+  final PhotoViewController photoViewController = PhotoViewController();
 
-  const ImageViewer({super.key, required this.url});
+  ImageViewer({super.key, required this.url});
 
-  @override
-  State<ImageViewer> createState() => _ImageViewerState();
-}
+  _requestPermission() async {
+    Map<Permission, PermissionStatus> statuses = await [
+      Permission.photos,
+    ].request();
 
-class _ImageViewerState extends State<ImageViewer> {
-  GlobalKey<ExtendedImageSlidePageState> slidePagekey = GlobalKey<ExtendedImageSlidePageState>();
+    final info2 = statuses[Permission.photos].toString();
+    print(info2);
+  }
 
   @override
   Widget build(BuildContext context) {
+    final theme = Theme.of(context);
     return Scaffold(
-      backgroundColor: Colors.black87,
-      body: Center(
-        child: ExtendedImageSlidePage(
-          key: slidePagekey,
-          slideAxis: SlideAxis.both,
-          slideType: SlideType.onlyImage,
-          child: GestureDetector(
-            child: HeroWidget(
-              tag: widget.url,
-              slideType: SlideType.onlyImage,
-              slidePagekey: slidePagekey,
-              child: ExtendedImage.network(
-                widget.url,
-                enableSlideOutPage: true,
-                mode: ExtendedImageMode.gesture,
-                initGestureConfigHandler: (ExtendedImageState state) {
-                  return GestureConfig(
-                    minScale: 0.9,
-                    animationMinScale: 0.7,
-                    maxScale: 4.0,
-                    animationMaxScale: 4.5,
-                    speed: 1.0,
-                    inertialSpeed: 100.0,
-                    initialScale: 1.0,
-                    inPageView: false,
-                    initialAlignment: InitialAlignment.center,
-                    reverseMousePointerScrollDirection: true,
-                    gestureDetailsIsChanged: (GestureDetails? details) {},
-                  );
+      body: SafeArea(
+        child: Stack(
+          alignment: Alignment.topRight,
+          children: [
+            PhotoView(
```

Comment:
hmm, this was changed by me a couple days ago. In image_viewer, I'm no longer using the `PhotoView` widget - but rather `ExtendedImageSlidePage`

Can you double check to see if maybe there was a commit which forced this change back to what it was previously?